src/components/TodoItem.tsx

```diff
@@ -0,0 +1,198 @@
+import { FormEvent, useEffect, useState } from 'react';
+import cn from 'classnames';
+import { TContext, useTodoContext } from '../context/TodoContext';
+import { Todo } from '../types/Todo';
+import { deleteTodo, editTodo, getTodos } from '../api/todos';
+
+type Props = {
+  todo: Todo,
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const [isDeleting, setIsDeleting] = useState<boolean>(false);
+  const [toggledId, setToggledId] = useState<number | null>(null);
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const [isLoading, setIsLoading] = useState(false);
+  const [editedTodo, setIsEditedTodo] = useState<Todo | null>(null);
+
+  const {
+    todos,
+    setTodos,
+    handleError,
+    handleToggleStatus,
+    isToggled,
+    isToggledAll,
+    titleInputRef,
+    isGroupDeleting,
+    editedRef,
+  } = useTodoContext() as TContext;
+
+  useEffect(() => {
+    if (isEditing) {
+      editedRef.current?.focus();
+    }
+  }, [isEditing]);
+
+  const USER_ID = 11550;
+
+  const isActiveConditions = (isDeleting === true)
+  || (todo.id === 0) || (isToggled && todo.id === toggledId)
+  || (isToggledAll) || (isLoading)
+  || ((todo.completed) && isGroupDeleting);
+
```

Comment:
It's good that you extracted it outside JSX, however this logic still causes some of the tests to fail. Try to fix it a bit. It should not be dependent on so many cases.